doc/source/rllib/package_ref/trainer.rst

```diff
@@ -0,0 +1,50 @@
+.. _trainer-reference-docs:
+
+Trainer API
+===========
+
+The ``Trainer`` class is the highest-level API in RLlib. It allows you to
+train and evaluate policies, save an experiment's progress and restore from
+a prior saved experiment when continuing an RL run. ``Trainer`` is a sub-class
+of ``tune.Trainable`` and thus fully supports distributed hyperparameter tuning
+for RL.
+
+.. https://docs.google.com/drawings/d/1J0nfBMZ8cBff34e-nSPJZMM1jKOuUL11zFJm6CmWtJU/edit
+.. figure:: ../../images/rllib/trainer_class_overview.svg
+    :align: left
+
+    **A typical RLlib Trainer object:** The components sitting inside a Trainer are
+    normally: N ``WorkerSets`` (each consisting of one local ``RolloutWorker`` and zero or
+    more @ray.remote ``RolloutWorkers``),
+    a set of ``Policy/ies`` and their NN models per worker, and a (already vectorized)
+    RLlib ``BaseEnv`` per worker.
+
+
+Building Custom Trainer Classes
++++++++++++++++++++++++++++++++
+
+.. warning::
+    It is not recommended anymore to use the `build_trainer()` utility function
+    for creating custom Trainer sub-classes.
```

Comment:
This is old right? Can you say "As of X version of Ray, it is not recommended..."

Reply:
done